CMSgov/easi-app
# [EASI-3021] CSV Portfolio Update Report

# EASI-3021

[Figma](https://www.figma.com/file/HR3gewBLkFhR6gQS2RUKdC/IT-Governance-Export-and-Reporting?type=design&node-id=74%3A2309&mode=design&t=TMRiZc5CJAIuPACQ-1)

## Changes and Description

- Created `portfolioUpdateReport` feature flag
- Split existing download requests link into separate links for open or closed
- Portfolio Update Report modal
  - "Configure Portfolio Update Report" button triggers modal
  - Created modal for date range filtering and downloading report ([Figma](https://www.figma.com/file/HR3gewBLkFhR6gQS2RUKdC/IT-Governance-Export-and-Reporting?type=design&node-id=74%3A2309&mode=design&t=ydRNz8Jr4s3t8V4W-1))
- Portfolio Update Report download
  - Only includes closed requests
  - Updated column names and order (see file in Jira ticket for reference)
    -  New csv header map file: src/components/RequestRepository/csvPortfolioReportHeaderMap.ts
  - Filtering is based on date of last admin note. If intake does NOT have admin notes, filtering uses date of last admin action. [(Code to get filter date is here)](https://github.com/CMSgov/easi-app/blob/1db91bb005180bd3cd46fe96857ea66aaa4cde25/src/components/RequestRepository/tableMap.ts#L83-L94)
- Replaced redux with gql query (getSystemIntakesTable) to get system intakes for table
  - Updated cypress tests to remove redux references
- Backend work
  - Created systemIntakes query
  - Removed lastAdminNote field from systemIntake (calculated on frontend instead)



## How to test this change

- Log in as IT Gov admin with `portfolioUpdateReport` flag turned on
- Test Portfolio Update Report download
  - Unit test for date filtering: src/components/RequestRepository/tableMap.test.ts
  - Check that only closed requests are downloaded
  - Check that csv format matches "XXXXXX CMS IT Portfolio Updates Template.xlsx" template file attached to Jira ticket
- Test open and closed download buttons

 
## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
